staging/src/k8s.io/apiserver/pkg/endpoints/request/requestcontext.go

```diff
@@ -78,6 +78,10 @@ func (c *requestContextMap) Update(req *http.Request, context Context) error {
 // if a context was already associated with the request, it ignores the given context and returns false.
 // init is intentionally unexported to ensure that all init calls are paired with a remove after a request is handled
 func (c *requestContextMap) init(req *http.Request, context Context) bool {
+	if _, exists := c.Get(req); exists {
```

Comment:
since we only add the filter once, this actually is worse in the normal case of a single init() call per request, since we'll need to acquire the read lock, fail to find the existing entry, then acquire the write lock. I would remove this change and just keep the change in `Get`

Reply:
also, if you are looking for ways to improve contention on the map, avoiding `Update` needing to write-lock the map would be a much bigger win, since several filters update the context.

You could do that by changing the map to be map[request]atomic.Value, and have Update read-lock the map, get the value, and call Store() on the value.

Reply:
>since we only add the filter once, this actually is worse in the normal case of a single init() call per request

missed it, agree

Reply:
atomic.Value.Store(context) in Update paniced with 
`			panic("sync/atomic: store of inconsistently typed value into Value")
`
so this way is not viable

Reply:
In order to use atomic.Value, I wrap the Context interface. to prevent  `atomic.Value.Store` panic